app/renderer/src/store/tasks/index.ts

```diff
@@ -1,3 +1,322 @@
+        (card) => card._id === action.payload.cardId
+      );
+      if (taskIndex === -1) return state;
+      const oldTask = list.cards[taskIndex];
+
+      const newTask = editTask(oldTask, {
+        description: action.payload.description?.capitalize() || "",
+      });
+
+      state[listIndex].cards.splice(taskIndex, 1, newTask);
+    },
+
+    removeTaskCard: (
+      state,
+      action: PayloadAction<{
+        listId: TaskList["_id"];
+        cardId: Task["_id"];
+      }>
+    ) => {
+      const listIndex = state.findIndex(
+        (list) => list._id === action.payload.listId
+      );
+      if (listIndex === -1) return;
+      const oldTaskList = state[listIndex];
+
+      const newTaskList = removeTaskFromList(
+        oldTaskList,
+        action.payload.cardId
+      );
+
+      state[listIndex] = newTaskList;
+    },
+
+    setTaskCardDone: (
+      state,
+      action: PayloadAction<{
+        listId: TaskList["_id"];
+        cardId?: Task["_id"];
+      }>
+    ) => {
+      if (!action.payload.cardId) return;
+
+      const listIndex = state.findIndex(
+        (list) => list._id === action.payload.listId
+      );
+      if (listIndex === -1) return state;
+      const list = state[listIndex];
+
+      const taskIndex = list.cards.findIndex(
+        (card) => card._id === action.payload.cardId
+      );
+      if (taskIndex === -1) return;
+
+      const oldTask = list.cards[taskIndex];
+
+      const newTask = editTask(oldTask, { done: true });
+
+      state[listIndex].cards.splice(taskIndex, 1, newTask);
+    },
+
+    setTaskCardNotDone: (
+      state,
+      action: PayloadAction<{
+        listId: TaskList["_id"];
+        cardId?: Task["_id"];
+      }>
+    ) => {
+      if (!action.payload.cardId) return;
+
+      const listIndex = state.findIndex(
+        (list) => list._id === action.payload.listId
+      );
+      if (listIndex === -1) return;
+      const list = state[listIndex];
+
+      const taskIndex = list.cards.findIndex(
+        (card) => card._id === action.payload.cardId
+      );
+      if (taskIndex === -1) return;
+      const oldTask = list.cards[taskIndex];
+
+      const newTask = editTask(oldTask, { done: false });
+
+      state[listIndex].cards.splice(taskIndex, 1, newTask);
+    },
+
+    skipTaskCard: (state, action: TasksPayload<"_id">) => {
+      const listIndex = state.findIndex(
+        (list) => list._id === action.payload
+      );
+      if (listIndex === -1) return;
+      const list = state[listIndex];
+
+      if (list.cards.length === 0) return;
+
+      const notDoneCards = list.cards.filter((card) => !card.done);
+      const doneCards = list.cards.filter((card) => card.done);
+
+      const firstNotDoneCard = notDoneCards.at(0);
+
+      if (!firstNotDoneCard) return;
+
+      state[listIndex] = {
+        ...list,
+        cards: [
+          // Not done without the old first card
+          ...notDoneCards.splice(0, 1),
```

Comment:
Here is the error

splice returns the deleted one and not the rest of the Tasks